.github/workflows/format.sh

```diff
@@ -0,0 +1,4 @@
+#!/bin/bash
+
+make format
+test $(git status --porcelain | wc -l) -eq 0 || { git diff; false;  }
```

Comment:
Why not put this in the workflow? It seems consistent to create a script for these two lines.

Reply:
Do you mean "inconsistent"?  Anyways, I prefer to keep logic out of the github action YAML where possible.  It makes it easier to invoke locally, etc.

Reply:
Yes, but that is not done in other steps, such as https://github.com/SymbiFlow/python-fpga-interchange/pull/5/files#diff-b803fcb7f17ed9235f1e5cb1fcd2f5d3b2838429d4368ae4c57ce4436577f03fR30-R35.

Personally, I like the idea of putting the logic that could be CI independent in either a Makefile or a single script, like a CI abstraction layer.